### PR TITLE
[KeyVault] KeyVault Round 3 Commands

### PR DESCRIPTION
**Pull request by tjprescott**
This PR adds the following KeyVault commands and associated tests:

- **key**: import, backup, restore
- **certificate**: import (PEM and PFX support)
- **certificate pending**: show, merge, delete

### CODE REVIEW

**Comment by tjprescott** [comment]
**key import**
```
Command
    az keyvault key import

Arguments
    --name -n    [Required]: Name of the key.
    --vault-name [Required]: Name of the key vault.
    --disabled             : Create key in disabled state.
    --expires              : Expiration UTC datetime  (Y-m-d'T'H:M'Z').
    --not-before           : Key not usable before the provided UTC datetime  (Y-m-d'T'H:M'Z').
    --ops                  : Space separated list of permitted JSON web key operations. Possible
                             values: encrypt, decrypt, sign, verify, wrapKey, unwrapKey.
    --protection -p        : Specifies the type of key protection.  Allowed values: hsm, software.
    --tags                 : Space separated tags in 'key[=value]' format. Use "" to clear existing
                             tags.
```

**key backup**
```
Command
    az keyvault key backup: Requests that a backup of the specified key be downloaded to the client.

Arguments
    --file-path  [Required]
    --name -n    [Required]: Name of the key.
    --vault-name [Required]: Name of the key vault.
```

**key restore**
```
Command
    az keyvault key restore: Restores the backup key in to a vault.

Arguments
    --file-path  [Required]
    --vault-name [Required]: Name of the key vault.
```

**certificate import**
```
Command
    az keyvault certificate import: Imports a certificate into the specified vault.

Arguments
    --file -f    [Required]: PKCS12 file or PEM file containing the certificate and private key.
    --name -n    [Required]: Name of the certificate.
    --vault-name [Required]: Name of the key vault.
    --disabled             : Create certificate in disabled state.
    --expires              : Expiration UTC datetime  (Y-m-d'T'H:M'Z').
    --not-before           : Key not usable before the provided UTC datetime  (Y-m-d'T'H:M'Z').
    --password             : If the private key in base64EncodedCertificate is encrypted, the
                             password used for encryption.
    --policy -p            : JSON encoded policy defintion. Use @{file} to load from a file.
    --tags                 : Space separated tags in 'key[=value]' format. Use "" to clear existing
                             tags.
```

**certificate pending merge**
```
Command
    az keyvault certificate pending merge: Merges a certificate or a certificate chain with a key
    pair existing on the server.

Arguments
    --file -f    [Required]: File containing the certificate or certificate chain to merge.
    --name -n    [Required]: Name of the certificate.
    --vault-name [Required]: Name of the key vault.
    --disabled             : Create certificate in disabled state.
    --expires              : Expiration UTC datetime  (Y-m-d'T'H:M'Z').
    --not-before           : Key not usable before the provided UTC datetime  (Y-m-d'T'H:M'Z').
    --tags                 : Space separated tags in 'key[=value]' format. Use "" to clear existing
                             tags.
```

**Comment by tjprescott** [comment]
# Group Help

**keyvault key**
```
Group
    az keyvault key: Manage keys.

Commands:
    backup        : Requests that a backup of the specified key be downloaded to the client.
    create        : Creates a new, named, key in the specified vault.
    delete        : Deletes the specified key.
    import
    list          : List keys in the specified vault.
    list-versions : List the versions of the specified key.
    restore       : Restores the backup key in to a vault.
    set-attributes: Updates the Key Attributes associated with the specified key.
    show          : Retrieves the public portion of a key plus its attributes.
```

**keyvault certificate**
```
Group
    az keyvault certificate: Manage certificates.

Subgroups:
    contact       : Manage contacts for certificate management.
    issuer        : Manage certificate issuer information.
    pending       : Manage pending certificate creation operations.

Commands:
    create        : Creates a new certificate version.
    delete        : Deletes a certificate from the specified vault.
    import        : Imports a certificate into the specified vault.
    list          : List certificates in the specified vault.
    list-versions : List the versions of a certificate.
    set-attributes: Updates the attributes associated with the specified certificate.
    show          : Gets a Certificate.
```

**keyvault certificate pending**
```
Group
    az keyvault certificate pending: Manage pending certificate creation operations.

Commands:
    delete: Deletes the certificate operation.
    merge : Merges a certificate or a certificate chain with a key pair existing on the server.
    show  : Gets the certificate operation response.
```

**Comment by tjprescott** [comment]
Already see a few params that need help text and some command help cleanup.

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py [hunk 1]
```diff
@@ -208,53 +214,104 @@ def create_key(client, vault_base_url, key_name, destination, key_size=None, key
...
+        key_obj.t = byok_data
 
     return client.import_key(
         vault_base_url, key_name, key_obj, destination == 'hsm', key_attrs, tags)
 
+# pylint: disable=unused-argument
+def download_secret(client, vault_base_url, secret_name, file_path, file_encoding='utf8',
+                    secret_version='', decode_binary=None):
+    secret = client.keyvault.get_secret(vault_base_url, secret_name, secret_version)
+    raise CLIError('TODO: implement')
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/custom.py]: Just checking this is intended?

File: src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_validators.py [hunk 2]
```diff
@@ -61,6 +67,8 @@ def validate_key_type(ns):
             'hsm': 'RSA-HSM'
         }
         ns.destination = dest_to_type_map[ns.destination]
+        if ns.destination == 'RSA' and hasattr(ns, 'byok_file') and ns.byok_file:
+            raise CLIError('BYOK keys are hardward protected. Omit --protection')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/_validators.py]: should be hardware

**Pull request merged**